server/services/ApplicationsDeveloper.js

```diff
@@ -44,61 +45,75 @@ class ApplicationsDeveloper {
     return baseUrl;
   }
 
-  // Remeber to remove the temporary query string once API is refcatored not to require it when doing a "fetch all"
-  getApplicationList (environment) {
-    const options = Object.assign(this._getBaseOptions(), {
-      method: 'GET',
-      url: `${this.server.baseUrl[environment]}/applications/?items_per_page=20&start_index=0`
-    });
-    logger.info(`Service request to retrieve ${environment} applications list, with payload: `, options);
-    return this.request(options);
+  async getApplicationList (oauthToken, environment) {
+    const serverUrl = this.server.baseUrl[environment];
+    logger.info(`trying to retrieve application list from environment=[${environment}] using serverUrl=[${serverUrl}]`);
+    const client = APIClientHelper.getPrivateAPIClient(oauthToken, serverUrl);
+
+    // MVP pagination
+    const itemsPerPage = 20; // api currently limits to 20, this should be increased for MVP without pagination options in the UI
```

Comment:
Didn't we have the story to raise this to a million rather than implement an unpaged get all api endpoint.

Reply:
yes but the current implementation of the api threw errors with anything over 20 so when that story gets picked up it'll need to change there first